request_test.go

```diff
@@ -504,82 +504,212 @@ func TestRequest_URLConcatenate(t *testing.T) {
 
 	reporter := NewAssertReporter(t)
 
-	config1 := Config{
-		BaseURL:  "",
-		Client:   client,
-		Reporter: reporter,
-	}
-
-	config2 := Config{
-		BaseURL:  "http://example.com",
-		Client:   client,
-		Reporter: reporter,
+	cases := []struct {
+		name      string
+		baseURL   string
+		method    string
+		path      string
+		pathArgs  []interface{}
+		setupFunc func(request *Request)
+		fullURL   string
+	}{
+		{
+			name:    "empty url, empty path",
+			baseURL: "",
+			method:  "GET",
+			path:    "",
+			fullURL: "",
+		},
+		{
+			name:    "empty path",
+			baseURL: "http://example.com",
+			method:  "GET",
+			path:    "",
+			fullURL: "http://example.com",
+		},
+		{
+			name:    "slash empty path",
+			baseURL: "http://example.com/",
+			method:  "GET",
+			path:    "",
+			fullURL: "http://example.com/",
+		},
+		{
+			name:    "with path",
+			baseURL: "http://example.com",
+			method:  "GET",
+			path:    "path",
+			fullURL: "http://example.com/path",
+		},
+		{
+			name:    "with path, without slash",
+			baseURL: "http://example.com",
+			method:  "GET",
+			path:    "path",
+			fullURL: "http://example.com/path",
+		},
+		{
+			name:    "with path, with slash",
+			baseURL: "http://example.com",
+			method:  "GET",
+			path:    "/path",
+			fullURL: "http://example.com/path",
+		},
+		{
+			name:    "with path, without slash",
+			baseURL: "http://example.com/",
+			method:  "GET",
+			path:    "path",
+			fullURL: "http://example.com/path",
+		},
+		{
+			name:    "with path, with slash",
+			baseURL: "http://example.com/",
+			method:  "GET",
+			path:    "/path",
+			fullURL: "http://example.com/path",
+		},
+		{
+			name:     "with path arg",
+			baseURL:  "http://example.com/",
+			method:   "GET",
+			path:     "{arg}",
+			pathArgs: []interface{}{"/path"},
+			fullURL:  "http://example.com/path",
+		},
+		{
+			name:    "with setup func",
+			baseURL: "http://example.com/",
+			method:  "GET",
+			path:    "{arg}",
+			setupFunc: func(req *Request) {
+				req.WithPath("arg", "/path")
+			},
+			fullURL: "http://example.com/path",
+		},
 	}
 
-	config3 := Config{
-		BaseURL:  "http://example.com/",
-		Client:   client,
-		Reporter: reporter,
-	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			var req *Request
+
+			if tc.pathArgs != nil {
+				req = NewRequestC(
+					Config{
+						BaseURL:  tc.baseURL,
+						Client:   client,
+						Reporter: reporter,
+					},
+					tc.method,
+					tc.path,
+					tc.pathArgs...,
+				)
+			} else if tc.setupFunc != nil {
+				req = NewRequestC(
+					Config{
+						BaseURL:  tc.baseURL,
+						Client:   client,
+						Reporter: reporter,
+					},
+					tc.method,
+					tc.path,
+				)
+				tc.setupFunc(req)
+			} else {
+				req = NewRequestC(Config{
+					BaseURL:  tc.baseURL,
+					Client:   client,
+					Reporter: reporter,
+				}, tc.method, tc.path)
+			}
```

Comment:
We can simplify this branch:

First, unconditionally create req like this:

```go
				req = NewRequestC(
					Config{
						BaseURL:  tc.baseURL,
						Client:   client,
						Reporter: reporter,
					},
					tc.method,
					tc.path,
					tc.pathArgs...,
				)
```

(it should work if pathArgs is nil)

Then, check if setupFunc is non-nil, and call it.